File: carnival-search-server/src/providers/queues.ts
import { Queue } from 'bullmq';
import { config } from './config';
import redis from './redis';

export const index_queue = new Queue(config.index_queue, {
    connection: redis.duplicate()
})

export const add_vector_entity_queue = new Queue(config.add_vector_entity_queue, {
    connection: redis.duplicate()
})

export const delete_vector_entity_queue = new Queue(config.delete_vector_entity_queue, {
    connection: redis.duplicate()
})

File: carnival-search-vector-server/providers/queues.py
import os
from bullmq import Queue

# Redis connection configuration for Docker Compose
redis_connection = {
    "host": os.getenv("REDIS_HOST"),
    "port": int(os.getenv("REDIS_PORT")),
    "db": int(os.getenv("0"))
}

add_vector_entity_queue = Queue(os.getenv("ADD_VECTOR_ENTITY_QUEUE"), {"connection": redis_connection})
delete_vector_entity_queue = Queue(os.getenv("DELETE_VECTOR_ENTITY_QUEUE"), {"connection": redis_connection})
